### DISCUSSION TITLE
Windows7 PC freeze

### DISCUSSION THREAD

**Discussion by gsanchezv8** [arkypita/LaserGRBL#1235] [Q&A]
How are you guys? I have had compatibility problems for a long time, I have downloaded all versions, drivers, firewar and everything is the same ... when I run Laser GRBL on my win 7 the whole PC freezes, any solution?

**Comment by arkypita** [comment]
1st install or upgrade your Microsoft.net framework (Google for that)

2nd check if your antivirus is conflicting, try disable

No known issue with W7

**Reply by gsanchezv8** [reply]
I reinstalled Microsoft.NET and it remains the same, my pc freezes when running, and I don't have an antivirus

**Comment by StuartB4** [comment]
Have you tried a different usb port? May be a problem with that.

**Comment by arkypita** [comment]
What do you mean with "when I run"? Do you mean when you just open it, or when you are streaming a job to the machine?

What do you mean with "my pc freezes when running" exactly?
- does your mouse move or not?
- are you able to click on windows "start" menu? does it respond?
- are you able to move/minimize LaserGRBL window or not?
- are you able to move/minimize other programs or not?
- what does task manager show as CPU/RAM usage per process?

Please copy a screenshot when is possible to see LaserGRBL CPU/RAM usage like this:

![image](https://user-images.githubusercontent.com/8782035/103270793-14b33b80-49b9-11eb-90d7-0d6b0700254b.png)

**Reply by gsanchezv8** [reply]
Hello, my pc freezes and nothing works, not even the mouse. I updated my pc but it remains the same, I have 2gb of RAM but I want to clarify that Corel Draw, Aspire Vectric, ART Cam and Open Builds controller work perfectly on this PC.
I downloaded Laser GRBL on 4 different PCs, and it works perfectly!

**Comment by arkypita** [comment]
RAM is not a limit for lasergrbl, as it need only few amount.
Please respond to other questions:

What do you mean with "when I run"? Do you mean when you just open it? When you connect? When you open an image file? Or when you are streaming a job to the machine?

**Reply by gsanchezv8** [reply]
When I try to open the program! I don't even see the presentation of the program

**Comment by arkypita** [comment]
I am going to control in source code what is done in the very first part of the program startup.
In the meanwhile, can you check if something change if run with XP compatibility, or with Administrator privileges?

http://pearsonsupport.helpserve.com/Knowledgebase/Article/View/210/21/how-to-install-or-run-a-program-in-compatibility-mode-in-windows-7

**Comment by arkypita** [comment]
More than 8000 LaserGRBL user (15%) is using it on W7, so I am I'm pretty sure when I say there's no known issues :-)
In any case, every PC is different, both in terms of hardware, both in terms of installed programs and drivers, and sometimes small incompatibilities arise.

We need to investigate....

The most important and "dangerous" stuff that LaserGRBL does when he start is to re-program the OS task scheduler to run at 1ms instead of the default of 15ms (this to have more precise time measurements). This involves the whole system and all the applications that run (even if in a transparent way).

**Comment by arkypita** [comment]
I have created a test-build of LaserGRBL executable that log the startup sequence.
This test-build also disable the scheduler re-programming, so we can test if it was the cause of your issue.

- Download the .zip file
- Extract LaserGRBL.exe from the .zip file
- Replace the LaserGRBL.exe in C:\Program Files (x86)\LaserGRBL\ with the downloaded one
- Run LaserGRBL

If it work... we have found the issue is task scheduler reprogramming!
If it doesn't work we can see in LaserGRBL log file located under C:\Users\<your user name>\AppData\Roaming\LaserGRBL\sessionlog.txt at what point it hang!

[LaserGRBL.zip](https://github.com/arkypita/LaserGRBL/files/5752160/LaserGRBL.zip)

**Reply by gsanchezv8** [reply]
I did the steps one by one and it did not work, it froze again! I don't know what happens :(

**Comment by arkypita** [comment]
Can you open file located under C:\Users\your user name\AppData\Roaming\LaserGRBL\sessionlog.txt and copy the last lines here? 
Note: AppData is an hidden folder, if you don't see you should enable "show hidden folder" in folder browser option

they should look like this:

```
29/12/2020 23:23:17.426	Program     	------- LaserGRBL v3.8.9 [64bit] START -------
29/12/2020 23:23:17.426	Startup     	Init Update System
29/12/2020 23:23:17.450	Startup     	Load statistics
29/12/2020 23:23:17.474	Startup     	Load custom buttons
29/12/2020 23:23:17.482	Startup     	Enable visual styles
29/12/2020 23:23:17.490	Startup     	Set CompatibleTextRendering Default
29/12/2020 23:23:17.490	Startup     	Set Culture
29/12/2020 23:23:17.490	Startup     	Set Clock Resolution (Skipped)
29/12/2020 23:23:17.490	Startup     	Run Mainform
29/12/2020 23:23:17.498	Startup     	Inizialize Mainform
29/12/2020 23:23:17.962	Startup     	Show splash screen
29/12/2020 23:23:20.134	Startup     	Create Core object
29/12/2020 23:23:20.158	Startup     	Listen to Sync
29/12/2020 23:23:20.166	Startup     	Set Core
29/12/2020 23:23:20.229	Startup     	Refresh Title and Schema
```

**Reply by gsanchezv8** [reply]
![Captura](https://user-images.githubusercontent.com/76712089/103318614-c0867680-4a0d-11eb-8524-76ff4bc566fb.PNG)

**Reply by gsanchezv8** [reply]
That is all that I have

**Comment by arkypita** [comment]
Not c:\archivos de programa\ but c:\users\ (don't know in spanish)

C:\Users\your user name\AppData\Roaming\LaserGRBL\sessionlog.txt

**Comment by arkypita** [comment]
If you don't know how to open AppData\Roaming folder just simply press WIN+R and type %APPDATA%

![image](https://user-images.githubusercontent.com/8782035/103319587-9dba8900-4a32-11eb-82c9-f32433b5f0e1.png)

**Reply by gsanchezv8** [reply]
![Captura1](https://user-images.githubusercontent.com/76712089/103320616-d9def100-4a14-11eb-87bd-9db0f7b4e0d2.PNG)

**Reply by gsanchezv8** [reply]
THERE I FOUND! YOU SAY TO COPY AND REPLACE THE LINES YOU SENT ME?

**Reply by arkypita** [reply]
No no, dont replace... It's only a log file...

**Reply by arkypita** [reply]
This show me that it stop between "initialize mainform" and "show splash screen"

**Reply by gsanchezv8** [reply]
What do you suggest then?

**Reply by arkypita** [reply]
I go to sleep :-)
Wait for more news tomorrow

**Reply by gsanchezv8** [reply]
I appreciate your time, you are the best! Goodnight!

**Comment by arkypita** [comment]
Hi
Your problem obviously seems related to the machine you are using LaserGRBL on. From the test I made you do it seems that the problem is related to the management of the forms (windows).

Unfortunately it is something I can do very little about.

I recently released [a new pre-release](https://github.com/arkypita/LaserGRBL/releases/tag/v3.8.10), in which I also made some changes to the way I open the forms. Can you try it? I don't have much hope that it will solve your problem.

I think you might have the same problem with any .net framework based application. Has this ever happened to you with other programs?
